docs/content/docs/start/tutorial.md

```diff
@@ -43,51 +47,51 @@ const root = document.getElementById("root")
+The first client to start the app creates the container. The creation section of the application starts with calling `createContainer` and passing in a schema that defines which shared objects will be available on the new `container`. After a new container is created, default data can be set on the shared objects before the container is attached to the Tinylicious service.
 
-The attach call returns the `id` of the container, which the app can later use to load this container. Once attached, any further changes to the shared objects, made by the rendered app, will be communicated to all collaborators.
+The `attach` call returns the `id` of the container, which the app running on other clients can later use to load this container. Once attached, any further changes to the shared objects, made by the rendered app, will be communicated to all clients and, hence, all collaborators.
 
 ```js
 const createNewDice = async () => {
     const { container } = await client.createContainer(containerSchema);
     // Set default data
-    container.initialObjects.diceMap.set("dice-value-key", 1);
+    container.initialObjects.diceMap.set(diceValueKey, 1);
     // Attach container to service and return assigned ID
-    const id = container.attach();
-    // load the dice roller
+    const id = await container.attach();
+    // Render the dice roller
     renderDiceRoller(container.initialObjects.diceMap, root);
     return id;
-  }
+}
 ```
 
 ### Loading an existing container
 
-Loading a container is more straightforward than creating a new one. When loading, the container already contains data, and is already attached, so those steps are irrelevant. You need only to pass the `id` of the container you wish to load in the `getContainer()` function along with the same schema used when creating the container.
+Loading a container is more straightforward than creating a new one. When loading, the container already contains data, and is already attached, so those steps are irrelevant. You need only to pass the `id` of the container you wish to load in the `getContainer()` function along with the same schema used when creating the container. The schema is needed because the data in the container is compressed, so it needs to know the schema to reconstruct it.
 
 ```js
 const loadExistingDice = async (id) => {
   const { container } = await client.getContainer(id, containerSchema);
   renderDiceRoller(container.initialObjects.diceMap, root);
 }
-
 ```
 
 ### Switching between loading and creating
 
 The application supports both creating a new container and loading an existing container using its `id`.
-To control which state the app is in, this sample app stores the container ID in the URL hash.
+Each client, at startup, needs to know if it is the first client, in which case it needs to create the container. Otherwise, simply loads an existing container. The sample app uses its own URL to find this out. The app stores the container ID in the URL hash.
```

Comment:
```suggestion
Each client, at startup, needs to know if it is the first client, in which case it needs to create the container.
Otherwise, it may simply load an existing container.
The sample app uses its own URL to find this out.
The app stores the container ID in the URL hash.
```